assets/src/hooks/usePersistedStateReducer.ts

```diff
@@ -48,6 +59,16 @@ const usePersistedStateReducer = (): [State, Dispatch] => {
     }
   }, [selectedRouteIds, ladderDirections, ladderCrowdingToggles])
 
+  useEffect(() => {
+    if (firstLoadDone && pendingRouteTabs !== undefined) {
+      putRouteTabs(pendingRouteTabs).then((routeTabs) => {
```

Comment:
**thought**: this might take design assets we don't have yet, but pursuant to our recent talk about error-handling, should we handle network errors here more gracefully? and/or do we need a spinner?

Reply:
Indeed, this is probably one of the first cases where we would definitely want to handle it. I'm not really sure about the design resources and exactly what should happen when the network operation fails, though - do we still make the update in the front-end but just warn the user of the failure? Don't make the update at all and just revert the state of the application to what it was before the user initiated the change? I'd like to have at least a high-level answer to these questions before proceeding.

Reply:
yeah I think the answer here is, that's something we should have built in from the design stage, so let's leave it for now until we approach the question of errors more systematically

Reply:
Agreed, we should bring it up but it's at the moment still basically a product question.